src/classes/query-builder.js

```diff
@@ -1,89 +1,136 @@
+const capitalise = word => {
+  return word.charAt(0).toUpperCase() + word.slice(1)
+}
+
 module.exports = class QueryFilter {
-  constructor (knex) {
-    this.knex = knex
+  constructor (queryBuilder, Model) {
+    this.Model = Model
+    this.queryBuilder = queryBuilder
+    this.depth = 0 // nested query depth
+  }
+
+  _orBuilder (array) {
```

Comment:
I was initially a little unsure about the underscore naming convention but I suppose it's the best option. I guess we probably should make it a convention project wide.